alicat/tcp.py

```diff
@@ -94,19 +94,158 @@ async def set_gas(self, gas):
         """Set the gas type.
 
         Args:
-            gas: The gas type, as a string. Supported gas types are:
+            gas: The gas type, as a string or integer. Supported gas types by string are:
                 'Air', 'Ar', 'CH4', 'CO', 'CO2', 'C2H6', 'H2', 'He', 'N2',
                 'N2O', 'Ne', 'O2', 'C3H8', 'n-C4H10', 'C2H2', 'C2H4',
                 'i-C2H10', 'Kr', 'Xe', 'SF6', 'C-25', 'C-10', 'C-8', 'C-2',
                 'C-75', 'A-75', 'A-25', 'A1025', 'Star29', 'P-5'
+
+                For the full gas table, please see page 52 of the controller manual here:
+                https://documents.alicat.com/manuals/DOC-MANUAL-MC.pdf
+
+                Gas mixes may only be called by their mix number.
         """
-        if gas not in self.gases:
-            raise ValueError("{} not supported!".format(gas))
-        command = '{addr}$${gas}\r'.format(addr=self.address,
-                                           gas=self.gases.index(gas))
+        if isinstance(gas, str) is False:
+            num = True
+        else:
+            num = False
+
+        if num is True:
```

Comment:
Same comments all apply here.

I'm thinking about rewriting this so we don't have to repeat ourselves in the code, but unfortunately should copy it all over for now.